flink-libraries/flink-table/src/test/java/org/apache/flink/api/java/table/test/StringExpressionsITCase.java

```diff
@@ -121,6 +120,66 @@ public void testNonWorkingSubstring2() throws Exception {
 		resultSet.collect();
 	}
 
+	@Test
+	public void testStringConcat() throws Exception {
+		ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+		TableEnvironment tableEnv = new TableEnvironment();
+
+		DataSet<Tuple2<String, Integer>> ds = env.fromElements(
+			new Tuple2<>("ABCD", 3),
+			new Tuple2<>("ABCD", 2));
+
+		Table in = tableEnv.fromDataSet(ds, "a, b");
+
+		Table result = in
+			.select("a + b + 42");
```

Comment:
`What happens if you do "42 + a" or even "42 + b + a"?`

Am able to make the above work. 

`a + 42.abs().cast(STRING)`

I think as you said 42.abs() also does not work leave alone the .CAST added to it. 
abs() does not work  because it is not added in the ExpressionParser.scala itself. Reading the comment on ImplicitExpressionOperations i can see that the expressionDsl.scala and ExpressionParser should be in sync. You want me to add support for abs in this patch only? Let me know.